rclcpp/test/rclcpp/test_qos_event.cpp

```diff
@@ -518,6 +570,11 @@ TEST_F(TestQosEvent, test_sub_matched_event_by_set_event_callback)
 
 TEST_F(TestQosEvent, test_pub_matched_event_by_option_event_callback)
 {
+  std::string rmw_implementation_str = std::string(rmw_get_implementation_identifier());
```

Comment:
matched events should definitely work else there is a bug in rmw_zenoh.

Reply:
these one is failing 
```
Expected equality of these values:
  s.total_count
    Which is: 5
  matched_expected_result.total_count
    Which is: 1

/home/ahcorde/ros2_rolling/src/ros2/rclcpp/rclcpp/test/rclcpp/test_qos_event.cpp:577: Failure
Expected equality of these values:
  s.total_count_change
    Which is: 5
  matched_expected_result.total_count_change
    Which is: 1

/home/ahcorde/ros2_rolling/src/ros2/rclcpp/rclcpp/test/rclcpp/test_qos_event.cpp:578: Failure
Expected equality of these values:
  s.current_count
    Which is: 3
  matched_expected_result.current_count
    Which is: 1

/home/ahcorde/ros2_rolling/src/ros2/rclcpp/rclcpp/test/rclcpp/test_qos_event.cpp:576: Failure
Expected equality of these values:
  s.total_count
    Which is: 6
  matched_expected_result.total_count
    Which is: 1

/home/ahcorde/ros2_rolling/src/ros2/rclcpp/rclcpp/test/rclcpp/test_qos_event.cpp:577: Failure
Expected equality of these values:
  s.total_count_change
    Which is: 1
  matched_expected_result.total_count_change
    Which is: 0

/home/ahcorde/ros2_rolling/src/ros2/rclcpp/rclcpp/test/rclcpp/test_qos_event.cpp:578: Failure
Expected equality of these values:
  s.current_count
    Which is: 2
  matched_expected_result.current_count
    Which is: 0
```

Reply:
If I execute just the failing test the error is different:

```bash
test_qos_event --gtest_filter=TestQosEvent.test_sub_matched_event_by_option_event_callback

Expected equality of these values:
  s.total_count
    Which is: 2
  matched_expected_result.total_count
    Which is: 1

/home/ahcorde/ros2_rolling/src/ros2/rclcpp/rclcpp/test/rclcpp/test_qos_event.cpp:647: Failure
Expected equality of these values:
  s.total_count_change
    Which is: 1
  matched_expected_result.total_count_change
    Which is: 0
```

Reply:
The `event_statuses_` variable is not resetted or properly set when the node is destroyed

Reply:
Something around this https://github.com/ros2/rmw_zenoh/blob/rolling/rmw_zenoh_cpp/src/detail/graph_cache.cpp#L1303-L1306 is wrong.

Change should be negative, I don't understand the absolute value here. @Yadunund ?

Reply:
subscribers are not updated properly.

subscription related with parameter_events are only increasing the number.

Reply:
See https://github.com/ros2/rclcpp/pull/2626/files#r1762095271.